### DISCUSSION TITLE
Remotely verify device configuration

### DISCUSSION THREAD

**Discussion by mgulick** [emporia-vue-local/esphome#111] [Q&A] [answered]
This is more of an esphome question than emporia vue specifically.  I just updated my emporia vue config to add the `flash_write_interval: "48h"` setting, and I also changed the external_components source from `github://flaviut/esphome@emporia-vue-2022.4.0` to `github://emporia-vue-local/esphome@dev`.  Since I'm using an arm64-based home assistant blue, I can't use HA directly to update the esphome configuration of the emporia vue.  Instead, I used esphome via docker to rebuild the configuration and install it to the device wirelessly.  This appeared to go fine.  I could see the message that it was uploading the new firmware and rebooting the device, and the emporia vue is online and sending data again.

Is there any way to verify that the updated configuration was actually sent to the device?  I don't see any kind of a dashboard or management interface on the device itself (e.g. via http://IP-OF-EMPORIA-VUE).  Esphome seems a little strange to me in that the device's config file is on my local desktop, and when I run the dashboard, it pulls the hostname of the device from that config file in order to connect remotely to the device.  However it just kind of assumes the config file I have matches the one on the device.  I would have more of a warm and fuzzy feeling if I could validate that the device contains the configuration that _I think_ I just uploaded to it.

I forgot the process I used to add the device to HA originally, but I think I basically copy/pasted my local `config.yml` into the "new device" button on esphome in HA.  After doing this, the device appeared in HA.  However I didn't use HA to compile, install, or update the firmware on the device (because it won't compile on ARM64).  So this is also just basically using the config file to extract the hostname of the device in order to connect to it and show me that its online.  So after updating the emporia via esphome in docker, my Esphome tab on HA still shows the old config file.  I can (and did) edit the config in HA and copy/paste my new config there, but this is just superficial since the config in HA doesn't actually control the device's configuration (i.e. I can't upload a new config to the device via HA because it can't compile on arm64).  It would be pretty great if esphome pulled the existing configs from the device directly.

**Comment by ajacques** [comment]
ESPHome firmware usually publishes the running configuration to the MQTT broker that you've configured upon startup if the `logger` component is enabled. Try tailing the logs on `{emporia-vue2}/debug` MQTT topic after rebooting the Emporia Vue2. It won't give a YAML output, but it can be used to see what values are being used.

**Reply by mgulick** [reply]
Thanks for the suggestion, however I don't have MQTT enabled as I use the native home assistant API.

**Comment by flaviut** [comment]
The logs in the esphome console print the current config.

You can also flip the power to the device and observe that the daily power gets reset

**Comment by zanix** [comment, marked as answer]
I have this in my config that outputs the current firmware and build date

```yaml

text_sensor:
  - platform: version
    name: "EmporiaVue ESPHome Version"
```

Then whenever I update, there is a sensor with the current firmware version and date.

`sensor.emporiavue_esphome_version`
![image](https://user-images.githubusercontent.com/1580378/215939556-26a01586-6e87-4cdc-a7c6-9c5141b19cea.png)


And the firmware string is updated on the device page.
![image](https://user-images.githubusercontent.com/1580378/215939843-8d575247-4417-482c-9edd-6d68b19a7a5b.png)

**Reply by mgulick** [reply]
Thanks @zanix! This does allow me to confirm that a new build has been installed on the device.
